tools/toolchains/__init__.py

```diff
@@ -648,18 +679,34 @@ def link_program(self, r, tmp_path, name):
             self.progress("link", name)
             self.link(elf, objects, libraries, lib_dirs, linker_script)
 
-        if bin and self.need_update(bin, [elf]):
-            needed_update = True
-            self.progress("elf2bin", name)
-            self.binary(r, elf, bin)
+        if ext != 'elf':
+            if self.config.has_regions:
+                filename = "{}_application.{}".format(name, ext)
+            else:
+                filename = "{}.{}".format(name, ext)
+            full_path = join(tmp_path, filename)
+            if full_path and self.need_update(full_path, [elf]):
+                needed_update = True
+                self.progress("elf2bin", name)
+                self.binary(r, elf, full_path)
+            if self.config.has_regions:
+                full_path, updatable = self._do_region_merge(name, full_path, ext)
+            else:
+                updatable = None
+        else:
+            full_path = None
+            updatable = None
 
+        if self._post_build_hook:
+            self.progress("post-build", name)
+            self._post_build_hook(self, r, elf, full_path)
         # Initialize memap and process map file. This doesn't generate output.
         self.mem_stats(mapfile)
 
         self.notify.var("compile_succeded", True)
         self.notify.var("binary", filename)
 
-        return full_path, needed_update
+        return full_path, updatable
```

Comment:
Can we drop `needed_update` completely then? Doesn't seem to be used in this function.

Also, I can't find a call in our code base that actually uses the `updatable` result from `link_program()`. Is it possible to remove it?

Reply:
Yes. We can remove that too. We may have to update the online compiler to use this new API.